backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java

```diff
@@ -1312,88 +1310,69 @@ public DatasetPaginationDTO findDatasets(
+            roleService.getResourceItems(
+                null,
+                !queryParameters.getDatasetIdsList().isEmpty()
+                    ? new HashSet<>(queryParameters.getDatasetIdsList())
+                    : Collections.emptySet(),
+                ModelDBServiceResourceTypes.REPOSITORY);
         accessibleDatasetIds =
-            roleService.getSelfDirectlyAllowedResources(
-                ModelDBServiceResourceTypes.REPOSITORY,
-                ModelDBActionEnum.ModelDBServiceActions.READ);
-        if (queryParameters.getDatasetIdsList() != null
-            && !queryParameters.getDatasetIdsList().isEmpty()) {
-          accessibleDatasetIds.retainAll(queryParameters.getDatasetIdsList());
-        }
-        // user is in his workspace and has no repositorys, return empty
-        if (accessibleDatasetIds.isEmpty()) {
-          return emptyPaginationDTO;
-        }
+            accessibleAllWorkspaceItems.stream()
+                .map(GetResourcesResponseItem::getResourceId)
+                .collect(Collectors.toSet());
 
-        List<String> orgIds =
+        List<String> orgWorkspaceIds =
             roleService.listMyOrganizations().stream()
-                .map(Organization::getId)
+                .map(Organization::getWorkspaceId)
                 .collect(Collectors.toList());
-        if (!orgIds.isEmpty()) {
-          finalPredicatesList.add(
-              builder.not(
-                  builder.and(
-                      repositoryRoot.get(ModelDBConstants.WORKSPACE_ID).in(orgIds),
-                      builder.equal(
-                          repositoryRoot.get(ModelDBConstants.WORKSPACE_TYPE),
-                          WorkspaceType.ORGANIZATION_VALUE))));
-        }
-      } else {
-        if (resourceVisibility.equals(ResourceVisibility.PRIVATE)) {
-          List<KeyValueQuery> workspacePredicates =
-              ModelDBUtils.getKeyValueQueriesByWorkspace(
-                  roleService, currentLoginUserInfo, workspaceName);
-          if (workspacePredicates.size() > 0) {
-            Predicate privateWorkspacePredicate =
-                builder.equal(
-                    repositoryRoot.get(ModelDBConstants.WORKSPACE_ID),
-                    workspacePredicates.get(0).getValue().getStringValue());
-            Predicate privateWorkspaceTypePredicate =
-                builder.equal(
-                    repositoryRoot.get(ModelDBConstants.WORKSPACE_TYPE),
-                    workspacePredicates.get(1).getValue().getNumberValue());
-            Predicate privatePredicate =
-                builder.and(privateWorkspacePredicate, privateWorkspaceTypePredicate);
-
-            finalPredicatesList.add(privatePredicate);
+        for (GetResourcesResponseItem item : accessibleAllWorkspaceItems) {
+          if (orgWorkspaceIds.contains(String.valueOf(item.getWorkspaceId()))) {
+            accessibleDatasetIds.remove(item.getResourceId());
```

Comment:
I remember having a note on this filtering being redundant, right?

Reply:
Yes i remember that comments but here i have to filter out org_id from the my accessible ids with collaborators entities. there is no any endpoint which returns me just own entities ids and shared entity ids excluding org_id. that's why i have added these filter. if i am getting wrong on this let me know.

Reply:
It's fine, let's just tag a todo

Reply:
Added TODO